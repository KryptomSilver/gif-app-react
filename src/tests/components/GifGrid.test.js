import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import React from "react";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Debe de evaluar el componente <GifGrid/>", () => {
  const category = "Categoria";
  test("Debe de mostrar el componente correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wraper = shallow(<GifGrid category={category} />);
    expect(wraper).toMatchSnapshot();
  });
  test("Debe de mostrar items cuando se cargan imagenes setFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://localhost/",
        title: "Cualquier Cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wraper = shallow(<GifGrid category={category} />);
    // expect(wraper).toMatchSnapshot();
    expect(wraper.find("p").exists()).toBe(false);
    expect(wraper.find("GifGridItem").length).toBe(gifs.length);
  });
});
