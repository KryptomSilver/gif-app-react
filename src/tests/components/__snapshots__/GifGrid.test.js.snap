// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de evaluar el componente <GifGrid/> Debe de mostrar el componente correctamente 1`] = `ShallowWrapper {}`;
